src/main/java/com/google/step/YOUR_PROJECT_NAME_HERE/servlets/SearchServlet.java

```diff
@@ -15,24 +15,99 @@
 package com.google.step.YOUR_PROJECT_NAME_HERE.servlets;
 
 import java.io.IOException;
-import java.util.Arrays;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.ServletException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.json.JSONObject;
 
 /** Servlet that handles searches. */
 @WebServlet("/search")
 public class SearchServlet extends HttpServlet {
 
+  private final HttpClient httpClient =
+      HttpClient.newBuilder().version(HttpClient.Version.HTTP_2).build();
+
+  private static String encodeValue(String value) {
+    try {
+      return URLEncoder.encode(value, StandardCharsets.UTF_8.toString());
+    } catch (UnsupportedEncodingException ex) {
+      throw new RuntimeException(ex.getCause());
+    }
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
-    // Set some data on the request for the template to access.
-    request.setAttribute("previousQuery", request.getParameter("q"));
-    request.setAttribute("results", Arrays.asList("Aaron", "Iris", "Kayla"));
-    // Forward the request to the template (which is a servlet itself).
-    request.getRequestDispatcher("WEB-INF/templates/search.jsp").forward(request, response);
+
+    String query = encodeValue(request.getParameter("q"));
+    List<String> allLinks = new ArrayList<>();
+
+    String w3GoogleCSE =
+        "https://www.googleapis.com/customsearch/v1?key=AIzaSyCSKQy6t7HgEc1zI2thwgGncFVzyB8zZ00&cx=010448377421452380243:pat2rmwjvb8&q=";
+
+    String stackGoogleCSE =
+        "https://www.googleapis.com/customsearch/v1?key=AIzaSyCSKQy6t7HgEc1zI2thwgGncFVzyB8zZ00&cx=010448377421452380243:6b8rs1ze5oy&q=";
+
+    String geeksGoogleCSE =
+        "https://www.googleapis.com/customsearch/v1?key=AIzaSyCSKQy6t7HgEc1zI2thwgGncFVzyB8zZ00&cx=010448377421452380243:ieq7z84z2qq&q=";
+
+    try {
+      String w3Link = getLink(w3GoogleCSE + query);
+      allLinks.add(w3Link);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+
+    try {
+      String stackLink = getLink(stackGoogleCSE + query);
+      allLinks.add(stackLink);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+
+    try {
+      String geeksLink = getLink(geeksGoogleCSE + query);
+      allLinks.add(geeksLink);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+
+    response.setContentType("text/html;");
+    response.getWriter().println(allLinks);
```

Comment:
i think we planned to return in json format

Reply:
I'm just returning the links right now as an array so it's visible what the links are, but once the link is retrieved it should go directly to the respective function for scraping and then that function should return a JSON object